service/src/main/java/com/github/defsignals/service/service/JmxService.java

```diff
@@ -0,0 +1,47 @@
+package com.github.defsignals.service.service;
+
+import org.springframework.stereotype.Service;
+
+import javax.management.*;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class JmxService {
+    public static final String URL = "service:jmx:rmi:///jndi/rmi://%s:%d/jmxrmi";
+
+    private final ConcurrentHashMap<HostPort, JMXConnector> connectionMap = new ConcurrentHashMap<>();
+
+    public <T> T getMxBean(String host, Integer port, Class<T> MXBeanClass,
+                           String mxBeanConstantName) throws IOException, MalformedObjectNameException {
+
+        var jmxConnector = getConnection(host, port);
```

Comment:
the `jmxConnector` local variable is used only once to obtain an `MBeanServerConnection` object, let's hide the connector behind the `JmxConnectionPool` bean.